fdbserver/DataDistribution.actor.cpp

```diff
@@ -4061,103 +4136,88 @@ ACTOR Future<std::vector<std::pair<StorageServerInterface, ProcessClass>>> getSe
 	return results;
 }
 
-// Create a transaction reading the value of `wigglingStorageServerKey` and update it to the next Process ID according
-// to a sorted PID set maintained by the data distributor. If now no storage server exists, the new Process ID is 0.
-ACTOR Future<Void> updateNextWigglingStoragePID(DDTeamCollection* teamCollection) {
-	state ReadYourWritesTransaction tr(teamCollection->cx);
-	state Value writeValue = ""_sr;
-	state const Key writeKey =
-	    wigglingStorageServerKey.withSuffix(teamCollection->primary ? "/primary"_sr : "/remote"_sr);
+// return the next ServerID in storageWiggler
+ACTOR Future<UID> getNextWigglingServerID(DDTeamCollection* teamCollection) {
+	state Optional<Value> localityKey;
+	state Optional<Value> localityValue;
+
+	// NOTE: because normal \xff/conf change through `changeConfig` now will cause DD throw `movekeys_conflict()` then
+	// recruit a new DD, we only need to read current configuration once
+	if (teamCollection->configuration.perpetualStorageWiggleLocality != "0") {
+		std::string& localityKeyValue = teamCollection->configuration.perpetualStorageWiggleLocality;
+		ASSERT(isValidPerpetualStorageWiggleLocality(localityKeyValue));
+		// get key and value from perpetual_storage_wiggle_locality.
+		int split = localityKeyValue.find(':');
+		localityKey = Optional<Value>(ValueRef((uint8_t*)localityKeyValue.c_str(), split));
+		localityValue = Optional<Value>(
```

Comment:
suggestion: for readability may be useful to document the "parsing format". Also, can we place some guards to check if the parsing did the needful?